pkg/registry/resolver.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	api "github.com/openshift/ci-tools/pkg/api"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+// Registry will hold all the registry information needed to convert between the
+// user provided configs referencing the registry and the internal, complete
+// represenation
+type Registry struct{}
+
+func (r Registry) Resolve(config api.MultiStageTestConfiguration) (types.TestFlow, []error) {
```

Comment:
Let's return one `error` -- we can use `errors.NewAggregate(many, separate, errors)` to coalesce many errors into one. If the downstream consumer is not interested in ever breaking this array of errors apart, we do not need to return more than one `error`.